sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -862,6 +879,13 @@ class Analyzer(
         }
 
       case u: UnresolvedRelation => resolveRelation(u)
+
+      case u @ UnresolvedTable(SessionCatalogAndIdentifier(catalog, ident)) =>
```

Comment:
we should follow the insert case here
```
case u @ UnresolvedTable(SessionCatalogAndIdentifier(catalog, ident)) =>
  val relation = lookupRelation(catalog, ident, recurse = false).getOrElse(u)
  EliminateSubqueryAliases(relation) match {
    case _: View =>
      u.failAnalysis(s"$ident is a temp view not table.")
    case _: UnresolvedTable => u
    case _ => ResolvedTable(catalog, ident, relation)
  }
```

Reply:
This needs to go thru `lookupRelation` as suggested, no?